src/uglify/index.js

```diff
@@ -14,77 +13,69 @@ try {
 } catch (e) { } // eslint-disable-line no-empty
 
 export default class {
-  constructor(parallel = {}) {
-    let options = parallel;
-    if (typeof parallel === 'boolean') {
-      options = { cache: parallel, workers: parallel };
-    }
-    const { cache, workers } = options;
-    this.cache = cache === true ? findCacheDir({ name: 'uglifyjs-webpack-plugin' }) : cache;
-    this.workers = workers === true ? os.cpus().length - 1 : Math.min(Number(workers) || 0, os.cpus().length - 1);
+  constructor(options = {}) {
+    const { cache, parallel } = options;
+    this.cacheDir = cache === true ? findCacheDir({ name: 'uglifyjs-webpack-plugin' }) : cache;
+    this.maxConcurrentWorkers = parallel === true ? os.cpus().length - 1 : Math.min(Number(parallel) || 0, os.cpus().length - 1);
```

Comment:
Is there a reason why only using cpu count - 1? Why not using all CPUs.

---

I would replace `Math.min(Number(parallel) || 0, os.cpus().length - 1)` with `Number(parallel)`. When the user chooses a worker count we should respect it, even if it's more than CPUs.

Reply:
@sokra `-1` - one for master other for childs

Reply:
Yep one core needs to be reserved for the main process running the worker farm